### DISCUSSION TITLE
How to override shallowAugmentedArrayKeys on a Term

### DISCUSSION THREAD

**Discussion by hitfactory** [statamic/cms#5242] [Q&A]
This kind of approach which works for Assets doesn't work for Terms. 

`$this->app->bind(
            \Statamic\Contracts\Taxonomies\Term::class,
            CustomTerm::class
 );`

```
use Statamic\Taxonomies\Term;

class CustomTerm extends Term {
    public function shallowAugmentedArrayKeys() {
    	return ['id', 'url', 'title', 'slug', 'some_string_field'];
    }
}```

The shallowAugmentedArrayKeys method is on  Statamic\Taxonomies\LocalizedTerm, but if I extend that I get the following binding error:

`Illuminate\Contracts\Container\BindingResolutionException
Unresolvable dependency resolving [Parameter #0 [ <required> $term ]] in class Statamic\Taxonomies\LocalizedTerm`

How can I override shallowAugmentedArrayKeys for terms?

**Comment by duncanmcclean** [comment]
I don't think this will work but it's worth a try I guess..

Instead of extending & binding to [`Term`](https://github.com/statamic/cms/blob/3.2/src/Taxonomies/Term.php), try doing it to [`LocalizedTerm`](https://github.com/statamic/cms/blob/3.2/src/Taxonomies/LocalizedTerm.php). 

Basically, the `Term` itself doesn't have any of the augmentation stuff on it, it's all on the localised version, the `LocalizedTerm`.

**Comment by hitfactory** [comment]
I had tried extending CustomTerm like so and binding to the same interface

```
use Statamic\Taxonomies\LocalizedTerm;
class CustomTerm extends LocalizedTerm {
    public function shallowAugmentedArrayKeys() {
    	return ['id', 'url', 'title', 'slug', 'some_string_field'];
    }
}
```

But then I get the following error

```
lluminate\Contracts\Container\BindingResolutionException
Unresolvable dependency resolving [Parameter #0 [ <required> $term ]] in class App\CustomTerm
```

So perhaps it's not possible yet?
